modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/Layout/index.js

```diff
@@ -75,9 +87,42 @@ const ActivationStatusLayout = ({
 									title={null}
 									truncate={false}
 								>
-									<StatusTag
-										currentStatus={activationStatus.id}
-									/>
+									{isVisible && (
+										<ModalDXPCActivationStatus
+											observer={observer}
+											onClose={onClose}
+											projectID={projectID}
+										/>
+									)}
+
+									<div className="align-items-center d-flex">
+										<StatusTag
+											currentStatus={activationStatus.id}
+										/>
+
+										{userAccount.isStaff && (
+											<ButtonDropDown
+												customDropDownButton={
+													<ButtonWithIcon
+														displayType="null"
+														small
+														symbol="caret-bottom"
+													/>
+												}
+												items={[
+													{
+														label: 'Set to Active',
+														onClick: () =>
+															setIsVisible(true),
+													},
+												]}
+												menuElementAttrs={{
+													className:
+														'p-0 cp-activation-key-icon rounded-xs',
+												}}
+											/>
+										)}
```

Comment:
Are you checking if the status is In Progress?
Maybe you should put this in [STATUS_TAG_TYPE_NAMES.inProgress].